From 934dafd8f16e8c4b2b8421ceace74c5ea364f781 Mon Sep 17 00:00:00 2001
From: Steve Lhomme <robux4@ycbcr.xyz>
Date: Wed, 26 Apr 2023 11:55:59 +0200
Subject: [PATCH] meson: allow forcing the path to look for libraries

compiler.find_library() doesn't use c/cpp_lib_args when looking for libraries.

The way to pass customized path to look for libraries not found in the system
is to give the path(es) manually to compiler.find_library().
---
 meson_options.txt    |  2 ++
 src/glsl/meson.build | 14 +++++++-------
 2 files changed, 9 insertions(+), 7 deletions(-)

diff --git a/meson_options.txt b/meson_options.txt
index 5b8113ae..6c38ceea 100644
--- a/meson_options.txt
+++ b/meson_options.txt
@@ -52,3 +52,5 @@ option('debug-abort', type: 'boolean', value: false,
        description: 'abort() on most runtime errors (only for debugging purposes)')
 
 option('python-path', type: 'string', description: 'Path to python3 executable')
+
+option('find_lib_path', type: 'string', description: 'Path to libraries')
diff --git a/src/glsl/meson.build b/src/glsl/meson.build
index 5cebfb89..5e3ef2d5 100644
--- a/src/glsl/meson.build
+++ b/src/glsl/meson.build
@@ -31,7 +31,7 @@ elif not glslang_req.disabled()
     static   = arg[0]
     required = arg[1]
 
-    spirv = cxx.find_library('SPIRV', required: required, static: static)
+    spirv = cxx.find_library('SPIRV', required: required, static: static, dirs: [get_option('find_lib_path')])
 
     if not spirv.found()
       continue
@@ -42,13 +42,13 @@ elif not glslang_req.disabled()
     if static
       glslang_deps += [
         # Always required for static linking
-        cxx.find_library('MachineIndependent', required: true, static: true),
-        cxx.find_library('OSDependent',        required: true, static: true),
-        cxx.find_library('OGLCompiler',        required: true, static: true),
-        cxx.find_library('GenericCodeGen',     required: true, static: true),
+        cxx.find_library('MachineIndependent', required: true, static: true, dirs: [get_option('find_lib_path')]),
+        cxx.find_library('OSDependent',        required: true, static: true, dirs: [get_option('find_lib_path')]),
+        cxx.find_library('OGLCompiler',        required: true, static: true, dirs: [get_option('find_lib_path')]),
+        cxx.find_library('GenericCodeGen',     required: true, static: true, dirs: [get_option('find_lib_path')]),
         # SPIRV-Tools are required only if optimizer is enabled in glslang build
-        cxx.find_library('SPIRV-Tools',        required: false, static: true),
-        cxx.find_library('SPIRV-Tools-opt',    required: false, static: true),
+        cxx.find_library('SPIRV-Tools',        required: false, static: true, dirs: [get_option('find_lib_path')]),
+        cxx.find_library('SPIRV-Tools-opt',    required: false, static: true, dirs: [get_option('find_lib_path')]),
       ]
     endif
 
-- 
2.37.3.windows.1

